test/rulesets/Base/base.jl

```diff
@@ -174,4 +174,21 @@
         test_frule(Base.literal_pow, ^, 3.5, Val(3))
         test_rrule(Base.literal_pow, ^, 3.5, Val(3))
     end
+
+    @testset "Float conversions" begin
+        for f in (float, Float32, Float64, AbstractFloat)
+            test_frule(f, 1.2; rtol=1.0e-3, atol=1.0e-3)
+            test_frule(f, 1⊢1.7; rtol=1.0e-3, atol=1.0e-3)
+            test_rrule(f, 1.2; rtol=1.0e-3, atol=1.0e-3)
+
+            # test_rrule doesn't like integers, so test that case manually
+            let y = rand(Int), x = randn()
```

Comment:
WHy are bothinging with this rand stuff?
Why not
```julia
@test rrule(f, 9)[2](5.1)[2] == 5.1
```

Reply:
That'd be fine. I just like using random variables. It's like the poor man's symbolics.